File: 45.主元素/Solution.cpp
class Solution {
public:
    /**
     * @param nums: A list of integers
     * @return: The majority number
     */
    int majorityNumber(vector<int> nums) {
        // write your code here
        int rlt = nums[0]; 
        int n = 0;
        for(int i = 0; i < nums.size(); i++){  
            if(rlt == nums[i])  n++; else n--;
            if (n == 0) rlt = nums[i+1];
        }  
        return rlt; 
    }
};

File: 45.主元素/Solution.py
class Solution:
    """
    @param nums: A list of integers
    @return: The majority number
    """
    def majorityNumber(self, nums):
        # write your code here
        rlt = nums[0]
        count = 0
        for i, v in enumerate(nums):
            if v == rlt:
                count += 1
            else:
                count -= 1
                if count == 0:
                    rlt = nums[i+1]
        return rlt
